src/Migrate/deploymentscripts/Initialize-AzMigrateReplicationInfrastructure.ps1

```diff
@@ -0,0 +1,386 @@
+            Set-AzKeyVaultManagedStorageSasDefinition -AccountName $GateWayStorageAcName -VaultName $KeyVaultName -Name $gatewayStorageAccountSasSecretName -TemplateUri $gwyat -SasType 'account' -ValidityPeriod ([System.Timespan]::FromDays(30))
+            
+            # Phase 2
+           
+            # ServiceBusConnectionString
+            $serviceBusConnString = "ServiceBusConnectionString"
+            $serviceBusSecretObject = Get-AzKeyVaultSecret -VaultName $KeyVaultName -Name $serviceBusConnString -ErrorVariable notPresent -ErrorAction SilentlyContinue
+            if($serviceBusSecretObject){
+                Write-Host $serviceBusConnString, " for ", $applianceName, $LogStringSkipping
+            }else{
+                $serviceBusRootKey = Get-AzServiceBusKey -ResourceGroupName $ResourceGroupName -Namespace $ServiceBusNamespace -Name "RootManageSharedAccessKey"
+                $secret = ConvertTo-SecureString -String $serviceBusRootKey.PrimaryConnectionString -AsPlainText -Force
+                Set-AzKeyVaultSecret -VaultName $KeyVaultName -Name $serviceBusConnString -SecretValue $secret
+                Write-Host $LogStringCreated, $serviceBusConnString, " for ", $applianceName
+            }  
+
+            # Policy
+            $policyName = $MigratePrefix + $SiteName + "policy"
+            $existingPolicyObject = Get-AzMigrateReplicationPolicy -PolicyName $policyName -ResourceGroupName $ResourceGroupName -ResourceName $VaultName -ErrorVariable notPresent -ErrorAction SilentlyContinue
+            if(!$existingPolicyObject){
+                $providerSpecificPolicy = [Microsoft.Azure.PowerShell.Cmdlets.Migrate.Models.Api20180110.VMwareCbtPolicyCreationInput]::new()
+                $providerSpecificPolicy.AppConsistentFrequencyInMinute = 240
+                $providerSpecificPolicy.InstanceType = "VMwareCbt"
+                $providerSpecificPolicy.RecoveryPointHistoryInMinute = 4320
+                $providerSpecificPolicy.CrashConsistentFrequencyInMinute = 60
+                $existingPolicyObject = New-AzMigrateReplicationPolicy -PolicyName $policyName -ResourceGroupName $ResourceGroupName -ResourceName $VaultName -ProviderSpecificInput $providerSpecificPolicy
+                Write-Host $LogStringCreated, $policyName
+            }else{
+                Write-Host $policyName, $LogStringSkipping
+            }
+
+            # Policy-container mapping
+            $mappingName = "containermapping"
+            $allFabrics = Get-AzMigrateReplicationFabric -ResourceGroupName $ResourceGroupName -ResourceName $VaultName
+            foreach($fabric in $allFabrics){
+                if($fabric.Name -match $applianceName){
+                    $peContainers = Get-AzMigrateReplicationProtectionContainer -FabricName $fabric.Name -ResourceGroupName $ResourceGroupName -ResourceName $VaultName
+                    foreach ($peContainer in $peContainers) {
```

Comment:
under vmware v2 fabric there will be only one container. you can consider directly first instance, no need of looping and appliance name comparision.